galsim/real.py

```diff
@@ -457,8 +466,17 @@ def __init__(self, file_name=None, sample=None, image_dir=None, dir=None, preloa
         self.variance = self.cat.field('noise_variance') # noise variance for image
         self.mag = self.cat.field('mag')   # apparent magnitude
         self.band = self.cat.field('band') # bandpass in which apparent mag is measured, e.g., F814W
-        self.weight = self.cat.field('weight') # weight factor to account for size-dependent
-                                               # probability
+        # Add the weight factor to the catalog only if it behaves properly.  In particular, it
+        # should be a float between 0 and 1 (so that random selections of indices can use it to
+        # remove any selection effects in the catalog creation process).  If the weight does not
+        # behave like this, then the catalog simply won't have a weight column.
+        weight = self.cat.field('weight')
+        if np.min(weight) >= 0. and np.max(weight) <= 1. and \
+                not np.any(np.isnan(weight)) and not np.any(np.isinf(weight)):
```

Comment:
I'm not sure this whole if statement is required.  Currently, codecov tells us that we only ever do the `True` branch here.  So I'd suggest just always dividing by the max weight, rather than doing this test.  And just wherever we document how users can make their own catalog (Do we have such instructions?) we say that they should be float values > 0 and when used, they will be normalized by the maximum weight value.